odbc/src/api/functions.rs

```diff
@@ -337,13 +337,82 @@ fn sql_alloc_handle(
 #[no_mangle]
 pub unsafe extern "C" fn SQLBindCol(
     hstmt: HStmt,
-    _col_number: USmallInt,
-    _target_type: SmallInt,
-    _target_value: Pointer,
-    _buffer_length: Len,
-    _length_or_indicatior: *mut Len,
+    col_number: USmallInt,
+    target_type: SmallInt,
+    target_value: Pointer,
+    buffer_length: Len,
+    length_or_indicator: *mut Len,
 ) -> SqlReturn {
-    unimpl!(hstmt);
+    panic_safe_exec_keep_diagnostics!(
+        debug,
+        || {
+            let mongo_handle = MongoHandleRef::from(hstmt);
+            let stmt = must_be_valid!((*mongo_handle).as_statement());
+
+            // Currently, we only support column binding of one row at a time.
+            if stmt.attributes.read().unwrap().row_array_size != 1
+                || !stmt
+                    .attributes
+                    .read()
+                    .unwrap()
+                    .row_bind_offset_ptr
+                    .is_null()
+                || stmt.attributes.read().unwrap().row_bind_type
+                    != (BindType::SQL_BIND_BY_COLUMN as usize)
+            {
+                unsupported_function!(hstmt)
+            }
+
+            // makes sure that col_number is in bounds
+            let mongo_stmt = stmt.mongo_statement.read().unwrap();
+            if mongo_stmt.is_none() {
+                stmt.errors.write().unwrap().push(ODBCError::NoResultSet);
+                return SqlReturn::ERROR;
+            }
+
+            let col_max_size = mongo_stmt.as_ref().unwrap().get_resultset_metadata().len();
+
+            // columns are 1-indexed as per the ODBC spec.
+            if (col_number as usize) > col_max_size || col_number == 0 {
+                let mongo_handle = MongoHandleRef::from(hstmt);
+                add_diag_info!(mongo_handle, ODBCError::InvalidColumnNumber(col_number));
+                return SqlReturn::ERROR;
+            }
+
+            if stmt.bound_cols.read().unwrap().is_none() {
+                *stmt.bound_cols.write().unwrap() = Some(HashMap::new());
+            }
+
+            // Unbind column if target_value is null
+            if target_value.is_null() {
+                stmt.bound_cols
+                    .write()
+                    .unwrap()
+                    .as_mut()
+                    .unwrap()
+                    .remove(&col_number);
+            }
+            // Bind column
```

Comment:
I believe this works for the rebinding case as well, since insert if a key exists should overwrite the older buffer info, maybe worth calling out in this comment